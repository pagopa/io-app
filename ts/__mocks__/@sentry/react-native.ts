export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
