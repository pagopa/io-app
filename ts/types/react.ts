import { Component, ComponentType, StatelessComponent } from "react";
import {
  NavigationInjectedProps,
  NavigationScreenProps
} from "react-navigation";

/**
 * Evaluates to the Props type of a React component
 */
export type ComponentProps<C> = C extends StatelessComponent<infer P1>
  ? P1
  : C extends Component<infer P2>
    ? P2
    : C extends ComponentType<infer P3> ? P3 : never;

/**
 * Infers the type of the navigation params of a component
 */
export type InferNavigationParams<
  C,
  P = ComponentProps<C>
> = P extends NavigationInjectedProps<infer N>
  ? N
  : P extends NavigationScreenProps<infer N1> ? N1 : never;
