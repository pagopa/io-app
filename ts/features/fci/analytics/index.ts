import { getType } from "typesafe-actions";
import { mixpanel, mixpanelTrack } from "../../../mixpanel";
import { Action } from "../../../store/actions/types";
import {
  fciLoadQtspClauses,
  fciLoadQtspFilledDocument,
  fciSignatureRequestFromId,
  fciStartRequest,
  fciSigningRequest,
  fciUpdateDocumentSignaturesRequest,
  fciClearStateRequest,
  fciPollFilledDocument
} from "../store/actions";
import { getNetworkErrorMessage } from "../../../utils/errors";
import { SignatureRequestDetailView } from "../../../../definitions/fci/SignatureRequestDetailView";
import { buildEventProperties } from "../../../utils/analytics";

export const trackFciDocOpening = (
  expire_date: SignatureRequestDetailView["expires_at"],
  total_doc_count: number,
  environment: string
) =>
  void mixpanelTrack(
    "FCI_DOC_OPENING",
    buildEventProperties("UX", "action", {
      expire_date,
      total_doc_count,
      environment
    })
  );

export const trackFciUserExit = (
  screen_name: string,
  environment: string,
  cta_id?: string
) =>
  void mixpanelTrack(
    "FCI_USER_EXIT",
    buildEventProperties("UX", "exit", {
      screen_name,
      cta_id,
      environment
    })
  );

export const trackFciUxConversion = (environment: string) =>
  void mixpanelTrack(
    "FCI_UX_CONVERSION",
    buildEventProperties("UX", "action", {
      environment
    })
  );

export const trackFciUserDataConfirmed = (environment: string) =>
  void mixpanelTrack(
    "FCI_USER_DATA_CONFIRMED",
    buildEventProperties("UX", "action", { environment })
  );

export const trackFciDocOpeningSuccess = (
  doc_count: number,
  sign_count: number,
  optional_sign_count: number,
  environment: string
) =>
  void mixpanelTrack(
    "FCI_DOC_OPENING_SUCCESS",
    buildEventProperties("UX", "control", {
      doc_count,
      sign_count,
      optional_sign_count,
      environment
    })
  );

export const trackFciSigningDoc = (environment: string) =>
  void mixpanelTrack(
    "FCI_SIGNING_DOC",
    buildEventProperties("UX", "action", { environment })
  );

export const trackFciShowSignatureFields = (environment: string) =>
  void mixpanelTrack(
    "FCI_SHOW_SIGNATURE_FIELDS",
    buildEventProperties("UX", "micro_action", { environment })
  );

export const trackFciUxSuccess = (
  doc_signed_count: number,
  signed_count: number,
  optional_signed_count: number,
  environment: string
) =>
  void mixpanelTrack(
    "FCI_UX_SUCCESS",
    buildEventProperties("UX", "screen_view", {
      doc_signed_count,
      signed_count,
      optional_signed_count,
      environment
    })
  );

export const trackFciStartSignature = (environment: string) =>
  void mixpanelTrack(
    "FCI_START_SIGNATURE",
    buildEventProperties("UX", "action", { environment })
  );

const trackFciAction =
  (mp: NonNullable<typeof mixpanel>, environment: string) =>
  (action: Action): Promise<void> => {
    switch (action.type) {
      case getType(fciStartRequest):
      case getType(fciSignatureRequestFromId.request):
      case getType(fciSignatureRequestFromId.success):
      case getType(fciLoadQtspClauses.request):
      case getType(fciLoadQtspClauses.success):
      case getType(fciLoadQtspFilledDocument.request):
      case getType(fciLoadQtspFilledDocument.success):
      case getType(fciSigningRequest.request):
      case getType(fciUpdateDocumentSignaturesRequest):
      case getType(fciClearStateRequest):
      case getType(fciPollFilledDocument.request):
      case getType(fciPollFilledDocument.success):
      case getType(fciPollFilledDocument.cancel):
        return mp.track(
          action.type,
          buildEventProperties("TECH", undefined, { environment })
        );
      case getType(fciSigningRequest.success):
        return mp.track(
          action.type,
          buildEventProperties("TECH", "control", { environment })
        );
      case getType(fciSignatureRequestFromId.failure):
      case getType(fciLoadQtspClauses.failure):
      case getType(fciLoadQtspFilledDocument.failure):
      case getType(fciSigningRequest.failure):
      case getType(fciPollFilledDocument.failure):
        return mp.track(
          action.type,
          buildEventProperties("KO", undefined, {
            reason: getNetworkErrorMessage(action.payload),
            environment
          })
        );
    }
    return Promise.resolve();
  };

export default trackFciAction;
