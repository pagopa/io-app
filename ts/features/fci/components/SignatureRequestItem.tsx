import * as React from "react";
import { View, StyleSheet } from "react-native";
import { HSpacer } from "@pagopa/io-app-design-system";
import { H4 } from "../../../components/core/typography/H4";
import { SignatureRequestListView } from "../../../../definitions/fci/SignatureRequestListView";
import { SignatureRequestStatusEnum } from "../../../../definitions/fci/SignatureRequestStatus";
import { IOBadge } from "../../../components/core/IOBadge";
import { LabelSmall } from "../../../components/core/typography/LabelSmall";
import TouchableDefaultOpacity from "../../../components/TouchableDefaultOpacity";
import ItemSeparatorComponent from "../../../components/ItemSeparatorComponent";
import I18n from "../../../i18n";

type Props = {
  item: SignatureRequestListView;
  onPress: () => void;
};

const styles = StyleSheet.create({
  badge: {
    alignContent: "center",
    alignItems: "center",
    justifyContent: "center"
  }
});

const SignatureRequestItem = (props: Props) => {
  const { item, onPress } = props;
  const renderStatusLabel = () => {
    switch (item.status) {
      case SignatureRequestStatusEnum.WAIT_FOR_SIGNATURE:
        return (
          <IOBadge
            small
            variant="solid"
            color="blue"
            text={I18n.t("features.fci.requests.signingInProgress")}
            testID={"FciSignatureRequestItemBadgeInProgress"}
          />
        );
      case SignatureRequestStatusEnum.SIGNED:
        return (
          <IOBadge
            small
            variant="solid"
            color="aqua"
            text={I18n.t("features.fci.requests.signed")}
            testID={"FciSignatureRequestItemBadgeSigned"}
          />
        );
      case SignatureRequestStatusEnum.WAIT_FOR_QTSP:
      default:
        return (
          <IOBadge
            small
            variant="outline"
            color="red"
            text={I18n.t("features.fci.requests.notAvailable")}
            testID={"FciSignatureRequestItemBadgeNotAvailable"}
          />
        );
    }
  };
  return (
    <View>
      <TouchableDefaultOpacity
        style={{ flexDirection: "row", paddingTop: 16, paddingBottom: 16 }}
        onPress={onPress}
        testID={"FciSignatureRequestOnPress"}
      >
        <View style={{ flexDirection: "column", flex: 1 }}>
          <H4>{item.dossier_title}</H4>
          <LabelSmall weight={"Regular"} color={"bluegrey"} numberOfLines={1}>
            {I18n.t("features.fci.requests.itemSubtitle", {
              date: item.created_at.toLocaleDateString(),
              time: item.created_at.toLocaleTimeString(),
              id: item.id
            })}
          </LabelSmall>
        </View>
        <HSpacer />
        <View style={styles.badge}>{renderStatusLabel()}</View>
      </TouchableDefaultOpacity>
      <ItemSeparatorComponent noPadded={true} />
    </View>
  );
};

export default SignatureRequestItem;
