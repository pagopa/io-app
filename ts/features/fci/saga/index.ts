import * as pot from "@pagopa/ts-commons/lib/pot";
import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
import { SagaIterator } from "redux-saga";
import { ActionType, isActionOf } from "typesafe-actions";
import RNFS from "react-native-fs";
import { call, takeLatest, put, select, take } from "typed-redux-saga/macro";
import { CommonActions, StackActions } from "@react-navigation/native";
import NavigationService from "../../../navigation/NavigationService";
import { FCI_ROUTES } from "../navigation/routes";
import ROUTES from "../../../navigation/routes";
import { apiUrlPrefix } from "../../../config";
import { SessionToken } from "../../../types/SessionToken";
import {
  identificationPinReset,
  identificationRequest,
  identificationSuccess
} from "../../../store/actions/identification";
import I18n from "../../../i18n";
import {
  fciSignatureRequestSelector,
  FciSignatureRequestState
} from "../store/reducers/fciSignatureRequest";
import { fciQtspFilledDocumentUrlSelector } from "../store/reducers/fciQtspFilledDocument";
import { CreateSignatureBody } from "../../../../definitions/fci/CreateSignatureBody";
import {
  fciSignatureRequestFromId,
  fciClearStateRequest,
  fciStartRequest,
  fciLoadQtspClauses,
  fciLoadQtspFilledDocument,
  fciDownloadPreview,
  fciDownloadPreviewClear,
  fciStartSigningRequest,
  fciSigningRequest,
  fciEndRequest,
  fciClearAllFiles,
  fciMetadataRequest,
  fciSignaturesListRequest,
  fciDocumentSignatureFields
} from "../store/actions";
import {
  fciQtspClausesMetadataSelector,
  FciQtspClausesState,
  fciQtspNonceSelector
} from "../store/reducers/fciQtspClauses";
import { fciDocumentSignaturesSelector } from "../store/reducers/fciDocumentSignatures";
import { KeyInfo } from "../../lollipop/utils/crypto";
import { createFciClient } from "../api/backendFci";
import { handleGetSignatureRequestById } from "./networking/handleGetSignatureRequestById";
import { handleGetQtspMetadata } from "./networking/handleGetQtspMetadata";
import { handleCreateFilledDocument } from "./networking/handleCreateFilledDocument";
import {
  FciDownloadPreviewDirectoryPath,
  handleDownloadDocument
} from "./networking/handleDownloadDocument";
import { handleCreateSignature } from "./networking/handleCreateSignature";
import { handleGetMetadata } from "./networking/handleGetMetadata";
import { handleGetSignatureRequests } from "./networking/handleGetSignatureRequests";
import { handleDrawSignatureBox } from "./handleDrawSignatureBox";

/**
 * Handle the FCI Signature requests
 * @param bearerToken
 */
export function* watchFciSaga(
  bearerToken: SessionToken,
  keyInfo: KeyInfo
): SagaIterator {
  const fciGeneratedClient = createFciClient(apiUrlPrefix);

  // handle the request of getting FCI signatureRequestDetails
  yield* takeLatest(
    fciSignatureRequestFromId.request,
    handleGetSignatureRequestById,
    fciGeneratedClient.getSignatureRequestById,
    bearerToken
  );

  // handle the request of getting QTSP metadata
  yield* takeLatest(
    fciLoadQtspClauses.request,
    handleGetQtspMetadata,
    fciGeneratedClient.getQtspClausesMetadata,
    bearerToken
  );

  // handle the request of getting QTSP filled_document
  yield* takeLatest(
    fciLoadQtspFilledDocument.request,
    handleCreateFilledDocument,
    fciGeneratedClient.createFilledDocument,
    bearerToken
  );

  yield* takeLatest(fciStartRequest, watchFciStartSaga);

  yield* takeLatest(fciLoadQtspClauses.success, watchFciQtspClausesSaga);

  // handle the request to get the document file from url
  yield* takeLatest(fciDownloadPreview.request, handleDownloadDocument);

  yield* takeLatest(fciDownloadPreviewClear, clearFciDownloadPreview);

  yield* takeLatest(fciStartSigningRequest, watchFciSigningRequestSaga);

  // handle the request to create the signature
  yield* takeLatest(
    fciSigningRequest.request,
    handleCreateSignature,
    apiUrlPrefix,
    bearerToken,
    keyInfo
  );

  yield* takeLatest(fciEndRequest, watchFciEndSaga);

  yield* takeLatest(fciClearAllFiles, clearAllFciFiles);

  yield* takeLatest(
    fciMetadataRequest.request,
    handleGetMetadata,
    fciGeneratedClient.getMetadata,
    bearerToken
  );

  yield* takeLatest(
    fciSignaturesListRequest.request,
    handleGetSignatureRequests,
    fciGeneratedClient.getSignatureRequests,
    bearerToken
  );

  yield* takeLatest(fciDocumentSignatureFields.request, handleDrawSignatureBox);

  yield* takeLatest(identificationPinReset, watchIdentificationPinResetSaga);
}

/**
 * Handle the identification pin reset to clear fci state
 */
function* watchIdentificationPinResetSaga(): SagaIterator {
  yield* put(fciClearStateRequest());
}

/**
 * Handle the FCI requests to get the QTSP filled_document
 */
function* watchFciQtspClausesSaga(): SagaIterator {
  const potQtspClauses: FciQtspClausesState = yield* select(
    fciQtspClausesMetadataSelector
  );

  if (pot.isSome(potQtspClauses)) {
    const documentUrl = Buffer.from(
      `${potQtspClauses.value.document_url}`
    ).toString("base64");
    yield* put(
      fciLoadQtspFilledDocument.request({
        document_url: documentUrl as NonEmptyString
      })
    );
  } else {
    yield* call(
      NavigationService.dispatchNavigationAction,
      CommonActions.navigate(ROUTES.MAIN, {
        screen: ROUTES.WORKUNIT_GENERIC_FAILURE
      })
    );
  }
}

/**
 * Handle the FCI start requests saga
 */
function* watchFciStartSaga(): SagaIterator {
  yield* call(
    NavigationService.dispatchNavigationAction,
    StackActions.replace(FCI_ROUTES.MAIN, {
      screen: FCI_ROUTES.DOCUMENTS,
      params: {
        attrs: undefined
      }
    })
  );
  // when the user start signing flow
  // start a request to get the QTSP metadata
  // this is needed to get the document_url
  // that will be used to create the filled document
  yield* put(fciLoadQtspClauses.request());

  // start a request to get the metadata
  // this is needed to get the service_id
  yield* put(fciMetadataRequest.request());
}

/**
 * Clears cached file for the fci document preview
 * and reset the state to empty.
 */
function* clearFciDownloadPreview(
  action: ActionType<typeof fciDownloadPreviewClear>
) {
  const path = action.payload.path;
  if (path) {
    yield* deletePath(path);
  }
  yield* put(fciDownloadPreview.cancel());
  yield* call(
    NavigationService.dispatchNavigationAction,
    CommonActions.goBack()
  );
}

/**
 * Handle the FCI start signing saga
 */
function* watchFciSigningRequestSaga(): SagaIterator {
  yield* put(
    identificationRequest(false, true, undefined, {
      label: I18n.t("global.buttons.cancel"),
      onCancel: () => undefined
    })
  );

  const res = yield* take(identificationSuccess);

  if (isActionOf(identificationSuccess, res)) {
    const potQtspClauses: FciQtspClausesState = yield* select(
      fciQtspClausesMetadataSelector
    );
    const potSignatureRequest: FciSignatureRequestState = yield* select(
      fciSignatureRequestSelector
    );
    const qtspFilledDocumentUrl = yield* select(
      fciQtspFilledDocumentUrlSelector
    );
    const qtspNonce = yield* select(fciQtspNonceSelector);
    const documentSignatures = yield* select(fciDocumentSignaturesSelector);

    if (
      pot.isSome(potQtspClauses) &&
      pot.isSome(potSignatureRequest) &&
      qtspFilledDocumentUrl &&
      qtspNonce
    ) {
      const createSignaturePayload: CreateSignatureBody = {
        signature_request_id: potSignatureRequest.value.id,
        documents_to_sign: documentSignatures,
        qtsp_clauses: {
          accepted_clauses: potQtspClauses.value.clauses,
          filled_document_url: Buffer.from(`${qtspFilledDocumentUrl}`).toString(
            "base64"
          ) as NonEmptyString,
          nonce: qtspNonce as NonEmptyString
        }
      };

      yield* put(fciSigningRequest.request(createSignaturePayload));
    }

    NavigationService.dispatchNavigationAction(
      CommonActions.navigate(FCI_ROUTES.MAIN, {
        screen: FCI_ROUTES.TYP
      })
    );
  }
}

function* deletePath(path: string) {
  yield RNFS.exists(path).then(exists =>
    exists ? RNFS.unlink(path) : Promise.resolve()
  );
}

/**
 * Clears cached file for the fci document preview
 * and reset the state to empty.
 */
function* clearAllFciFiles(action: ActionType<typeof fciClearAllFiles>) {
  yield* deletePath(action.payload.path);
}

/**
 * Handle the FCI abort requests saga
 */
function* watchFciEndSaga(): SagaIterator {
  yield* put(fciClearStateRequest());
  yield* put(fciClearAllFiles({ path: FciDownloadPreviewDirectoryPath }));
  yield* call(
    NavigationService.dispatchNavigationAction,
    CommonActions.navigate(ROUTES.MAIN)
  );
}
