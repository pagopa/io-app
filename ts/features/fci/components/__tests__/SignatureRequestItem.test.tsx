import * as React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import { SignatureRequestListView } from "../../../../../definitions/fci/SignatureRequestListView";
import SignatureRequestItem from "../SignatureRequestItem";
import { SignatureRequestStatusEnum } from "../../../../../definitions/fci/SignatureRequestStatus";
import { DossierTitle } from "../../../../../definitions/fci/DossierTitle";

type Props = {
  item: SignatureRequestListView;
  onPress: () => void;
};

const now = new Date();
const signatureRequestItem: SignatureRequestListView = {
  id: "mockedId" as SignatureRequestListView["id"],
  status: SignatureRequestStatusEnum.SIGNED,
  created_at: new Date(),
  dossier_id: "mockedDossierId" as SignatureRequestListView["dossier_id"],
  dossier_title: "mockedDossierTitle" as DossierTitle,
  expires_at: new Date(now.setDate(now.getDate() + 30)),
  signer_id: "mockedSignerId" as SignatureRequestListView["signer_id"],
  updated_at: new Date()
};

describe("Test SignatureRequestItem component", () => {
  it("should render a SignatureRequestItem component with props correctly", () => {
    const props = {
      item: signatureRequestItem,
      onPress: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
  });
  it("should render a SignatureRequestItem component with right title 'mockedDossierTitle'", () => {
    const props = {
      item: signatureRequestItem,
      onPress: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    expect(component.queryByText("mockedDossierTitle")).toBeTruthy();
  });
  it("should render a SignatureRequestItem component with signed label", () => {
    const item = signatureRequestItem;
    const props = {
      item: {
        ...item,
        status:
          SignatureRequestStatusEnum.SIGNED as SignatureRequestListView["status"]
      },
      onPress: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    expect(
      component.getByTestId("FciSignatureRequestItemBadgeSigned")
    ).toBeTruthy();
  });
  it("should render a SignatureRequestItem component with notAvailable label", () => {
    const item = signatureRequestItem;
    const props = {
      item: {
        ...item,
        status:
          SignatureRequestStatusEnum.WAIT_FOR_QTSP as SignatureRequestListView["status"]
      },
      onPress: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    expect(
      component.getByTestId("FciSignatureRequestItemBadgeNotAvailable")
    ).toBeTruthy();
  });
  it("should render a SignatureRequestItem component with inProgress label", () => {
    const item = signatureRequestItem;
    const props = {
      item: {
        ...item,
        status:
          SignatureRequestStatusEnum.WAIT_FOR_SIGNATURE as SignatureRequestListView["status"]
      },
      onPress: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    expect(
      component.getByTestId("FciSignatureRequestItemBadgeInProgress")
    ).toBeTruthy();
  });

  it("should render a SignatureRequestItem component and clickable", () => {
    const props = {
      item: signatureRequestItem,
      onPress: jest.fn()
    };
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
    const onTap = component.getByTestId("FciSignatureRequestOnPress");
    expect(onTap).toBeTruthy();
    fireEvent.press(onTap);
    fireEvent.press(onTap);
    expect(props.onPress).toHaveBeenCalledTimes(2);
  });
});

const renderComponent = (props: Props) =>
  render(<SignatureRequestItem {...props} />);
