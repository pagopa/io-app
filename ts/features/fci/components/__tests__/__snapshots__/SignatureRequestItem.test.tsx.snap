// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SignatureRequestItem component should render a SignatureRequestItem component with props correctly 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingTop": 16,
      }
    }
    testID="FciSignatureRequestOnPress"
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        color="bluegreyDark"
        font="TitilliumWeb"
        fontStyle={
          Object {
            "fontSize": 16,
            "lineHeight": 22,
          }
        }
        style={
          Array [
            Object {
              "fontSize": 16,
              "lineHeight": 22,
            },
            Object {
              "color": "#17324D",
              "fontFamily": "Titillium Web",
              "fontStyle": "normal",
              "fontWeight": "700",
            },
          ]
        }
        weight="Bold"
        weightColorFactory={[Function]}
      >
        mockedDossierTitle
      </Text>
      <Text
        color="bluegrey"
        defaultColor="blue"
        defaultWeight="Bold"
        font="TitilliumWeb"
        fontStyle={
          Object {
            "fontSize": 14,
          }
        }
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 14,
            },
            Object {
              "color": "#475A6D",
              "fontFamily": "Titillium Web",
              "fontStyle": "normal",
              "fontWeight": "400",
            },
          ]
        }
        weight="Regular"
      >
        4/13/2023·10:20:59 AM·ID mockedId
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 16,
        }
      }
    />
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#ED1727",
              "borderRadius": 13.5,
              "height": 27,
              "justifyContent": "center",
              "paddingBottom": 3,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 3,
            },
            Object {
              "elevation": 0.1,
              "paddingLeft": 8,
              "paddingRight": 8,
            },
            Object {
              "backgroundColor": "#0073E6",
            },
            null,
            Object {
              "height": 18,
            },
          ]
        }
        testID="FciSignatureRequestItemBadgeInProgress"
      >
        <Text
          color="white"
          defaultColor="blue"
          defaultWeight="Bold"
          font="TitilliumWeb"
          fontSize="small"
          fontStyle={
            Object {
              "fontSize": 12,
            }
          }
          style={
            Array [
              Object {
                "fontSize": 12,
              },
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
        >
          Signing in progress
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#CCD4DC",
        },
        Object {
          "marginLeft": 24,
          "marginRight": 24,
        },
        Object {
          "height": 0.5,
        },
      ]
    }
  />
</View>
`;
